tests/config_test.cc

```diff
@@ -170,14 +176,18 @@ TEST(config, consistent_toml_nonempty) {
 
 TEST(config, config_dirs_one_dir) {
   Config config(std::vector<boost::filesystem::path>{"tests/test_data/config_dirs/one_dir"});
-  EXPECT_EQ(config.storage.path, "/z_path");
+  EXPECT_EQ(config.storage.path.string(), "path_z");
+  EXPECT_EQ(config.pacman.sysroot.string(), "sysroot_z");
+  EXPECT_EQ(config.pacman.os, "os_a");
 }
 
 TEST(config, config_dirs_two_dirs) {
   std::vector<boost::filesystem::path> config_dirs{"tests/test_data/config_dirs/one_dir",
                                                    "tests/test_data/config_dirs/second_one_dir"};
   Config config(config_dirs);
-  EXPECT_EQ(config.storage.path, "/latest_path");
+  EXPECT_EQ(config.storage.path.string(), "path_z");
+  EXPECT_EQ(config.pacman.sysroot.string(), "sysroot_z");
+  EXPECT_NE(config.pacman.os, "os_a");
```

Comment:
This should be true, right? `pacman.os` doesn't get overwritten anywhere. The `one_dir` version of `a.conf` should get read and the `second_one_dir` skipped.

@cajun-rat Can you confirm that my understanding is correct?

Reply:
No, `a.conf` from `second_one_dir` should be read and should hide `a.conf` from `one_dir`